src/update_agent/mod.rs

```diff
@@ -35,17 +51,45 @@ enum UpdateAgentState {
 
 impl Default for UpdateAgentState {
     fn default() -> Self {
-        UpdateAgentState::StartState
+        let start_state = UpdateAgentState::StartState;
+        CURRENT_STATE.set(start_state.discriminant().into());
+        LATEST_STATE_CHANGE.set(chrono::Utc::now().timestamp());
+        start_state
     }
 }
 
 impl UpdateAgentState {
+    /// Return the discriminant for current state
+    fn discriminant(&self) -> u8 {
+        // TODO(lucab): update when arbitrary-discriminant is stabilized:
+        // https://github.com/rust-lang/rust/issues/60553
+
+        match self {
+            UpdateAgentState::StartState => 0,
+            UpdateAgentState::Initialized => 1,
+            UpdateAgentState::Steady => 2,
+            UpdateAgentState::UpdateAvailable(_) => 3,
+            UpdateAgentState::UpdateStaged(_) => 4,
+            UpdateAgentState::UpdateFinalized(_) => 5,
+            UpdateAgentState::EndState => 6,
+        }
```

Comment:
Hmm, does reporting these back to Prometheus imply we're stabilizing the various states we have? Or are we free to change these later on?

Reply:
Good question. The socket exposing this is still under `/private` so it's fine for now, but yes it will become part of the public interface and somehow stable. Maybe I should avoid exposing this.

Reply:
Probably not, though could we report strings instead? E.g. "initialized", "staged", "finalized", etc...

Reply:
We cannot **as values** (only f64 allowed). We can as part of **metrics name**, but that would crate a separate metric for each state, and monitoring folks strongly recommended against that. I'll just drop the state-id for now, I'll re-add later when the core logic is more stable.